src/github.com/matrix-org/dendrite/federationapi/routing/profile.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routing
+
+import (
+	"net/http"
+
+	"github.com/matrix-org/dendrite/clientapi/auth/storage/accounts"
+	"github.com/matrix-org/dendrite/clientapi/httputil"
+	"github.com/matrix-org/dendrite/clientapi/jsonerror"
+	clientRouting "github.com/matrix-org/dendrite/clientapi/routing"
+	"github.com/matrix-org/gomatrixserverlib"
+	"github.com/matrix-org/util"
+)
+
+// GetProfile implements GET /_matrix/federation/v1/query/profile
+func GetProfile(
+	httpReq *http.Request,
+	accountDB *accounts.Database,
+) util.JSONResponse {
+	userID, field := httpReq.FormValue("user_id"), httpReq.FormValue("field")
+
+	// httpReq.FormValue will return an empty string if value is not found
+	if userID == "" {
+		return util.JSONResponse{
+			Code: 400,
+			JSON: jsonerror.MissingArgument("The request body did not contain required argument 'user_id'."),
+		}
+	}
+
+	localpart, _, err := gomatrixserverlib.SplitID('@', userID)
```

Comment:
We probably want to check that the domain of the userID matches our server.

Reply:
Should that be done here or in `accountDB.GetProfileByLocalpart`?

Reply:
Should be done, check if my provided solution was correct though, not entirely sure myself...